### ISSUE TITLE
[BUG] ThreadStatic attribute causing EXCEPTION_ACCESS_VIOLATION

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#1290]
I'm not really familiar with this attribute but from my understanding this following code should work:
```cs
static class Foo
{
    [ThreadStatic]
    private static List<int32> _scopeInfo = new .() ~ delete _;

    internal static Dictionary<int, List<int32>> threads ~ delete _;
    internal static List<int> threadList ~ delete _;

    public static Result<void> Init()
    {
        threads = new .();
        threadList = new .();
        return InitThread(System.Threading.Thread.CurrentThread.Id);
    }

    static Result<void> InitThread(int32 threadId)
    {
        threads.Add(threadId, _scopeInfo);
        threadList.Add(threadId);
        return .Ok;
    }
}

class Program
{
    static void Main()
    {
        Foo.Init();
    }
}
``` 
But instead I get `EXCEPTION_ACCESS_VIOLATION reading from 0x140'272B9120 in thread 21380 at 0x7FF6'ED955B71`. If the [ThreadStatic] is removed it runs fine.

**Comment by disarray2077** [comment]
This error is because the static initializer only runs in the main thread, so `_scopeInfo` is undefined for every thread besides the main thread.

**Comment by bfiete** [comment]
Yes. This is how C# works, as well.

Also, there is (currently) no way to get a notification when a thread exits, so have to have a some of your own per-thread cleanup strategy if you're allocating per-thread.

**Comment by Fusioon** [comment]
but in the example code its only called from main thread. the application that this code is from does the allocation in InitThread and it still crashes.

**Comment by bfiete** [comment]
Ah yes. I did overlook the fact there's an actual bug here.

**Comment by bfiete** [comment]
Fixed at 74a63f7561f253ccb66f6e8f0320e576671b7d8c
